cli/cmd/sync_v3.go

```diff
@@ -326,10 +327,15 @@ func syncConnectionV3(ctx context.Context, source v3source, destinations []v3des
 			if err != nil {
 				return err
 			}
-			tableName := tableNameFromSchema(sc)
+			table, err := schema.NewTableFromArrowSchema(sc)
+			if err != nil {
+				return err
+			}
 
-			if !isStateBackendEnabled || !tableIsIncremental(sc) {
-				tablesForDeleteStale[tableName] = true
+			if isStateBackendEnabled && (table.IsIncremental || (table.Parent != nil && skippedFromDeleteStale[table.Parent.Name])) {
```

Comment:
IIRC the marshaled date will have only flattened stuff, so it's impossible to fix this on the destination side.

When running the sync we could to table -> schema.Tables{table).Flatten -> exclude all

Reply:
There's another fix in the SDK needed for this to send the parent in the migrate message which I still need to test https://github.com/cloudquery/plugin-sdk/compare/main...erezrokah:plugin-sdk:fix/send_parent_migrate?expand=1

Reply:
Works with https://github.com/cloudquery/plugin-sdk/pull/1742

Reply:
This at the very least implies that the parent migrate is sent before the relation migrate (which isn't documented anywhere as such).
Approving, but this should be taken into account & possibly documented as a requirement.